pkg/oci/client/oci.go

```diff
@@ -38,36 +39,54 @@ const (
 type Interface interface {
 	// FindVolumeAttachment searches for a volume attachment in either the state
 	// ATTACHING or ATTACHED and returns the first volume attachment found.
-	FindVolumeAttachment(volumeId string) (core.VolumeAttachment, error)
+	FindVolumeAttachment(volumeID string) (core.VolumeAttachment, error)
 
 	// WaitForVolumeAttached polls waiting for a OCI block volume to be in the
 	// ATTACHED state.
-	WaitForVolumeAttached(volumeAttachmentId string) (core.VolumeAttachment, error)
+	WaitForVolumeAttached(volumeAttachmentID string) (core.VolumeAttachment, error)
 
 	// GetInstanceByNodeName retrieves the oci.Instance corresponding or
 	// a SearchError if no instance matching the node name is found.
 	GetInstanceByNodeName(name string) (*core.Instance, error)
 
 	// AttachVolume attaches a block storage volume to the specified instance.
 	// See https://docs.us-phoenix-1.oraclecloud.com/api/#/en/iaas/20160918/VolumeAttachment/AttachVolume
-	AttachVolume(instanceId, volumeId string) (core.VolumeAttachment, int, error)
+	AttachVolume(instanceID, volumeID string) (core.VolumeAttachment, int, error)
 
 	// DetachVolume detaches a storage volume from the specified instance.
 	// See: https://docs.us-phoenix-1.oraclecloud.com/api/#/en/iaas/20160918/Volume/DetachVolume
-	DetachVolume(volumeAttachmentId string) error
+	DetachVolume(volumeAttachmentID string) error
 
 	// WaitForVolumeDetached polls waiting for a OCI block volume to be in the
 	// DETACHED state.
-	WaitForVolumeDetached(volumeAttachmentId string) error
+	WaitForVolumeDetached(volumeAttachmentID string) error
 
 	// GetConfig returns the Config associated with the OCI API client.
 	GetConfig() *Config
+
+	// GetMountTargetForAD returns a mount target for a given AD
+	GetMountTargetForAD(AvailabilityDomain string) (*ffsw.MountTarget, error)
+
+	// GetFilesystem
+	GetFileSystem(ocid string) (*ffsw.FileSystem, error)
+
+	// AttachFileSystemToMountTarget
+	AttachFileSystemToMountTarget(fileSystem *ffsw.FileSystem, mountTarget *ffsw.MountTarget, path string) error
+
+	// DetachFileSystemToMountTarget
+	DetachFileSystemToMountTarget(fileSystem *ffsw.FileSystem, mountTarget *ffsw.MountTarget, path string) error
+
+	// GetMountTargetIPS
+	GetMountTargetIPS(mountTarget *ffsw.MountTarget) ([]core.PrivateIp, error)
+
+	IsFileSystemAttached(fileSystem *ffsw.FileSystem, mountTarget *ffsw.MountTarget, path string) (bool, error)
 }
 
 // client extends a barmetal.Client.
```

Comment:
I guess its not a baremetal client anymore (although admittedly this is nothing to do with this change!)